bundle/mutator.go

```diff
@@ -34,28 +35,32 @@ func Apply(ctx context.Context, b *Bundle, m Mutator) error {
 		}
 	}()
 
-	err = m.Apply(ctx, b)
-	if err != nil {
+	diags := m.Apply(ctx, b)
+
+	// Log error in diagnostics if any.
+	// Note: errors should be logged when constructing them
+	// such that they are not logged multiple times.
+	// If this is done, we can omit this block.
+	if err := diags.Error(); err != nil {
```

Comment:
If I understand correctly by looking at `diag.FromErr`, `diags` can be nil if `err` passed is nil. In this case `diags.Error()` will wail, is it so?

Reply:
Nvm, it's a valid value for `Diagnostics` type so it works